interface/language/csv/validate_csv.php

```diff
@@ -0,0 +1,284 @@
+            </thead>
+            <tbody>
+                <?php
+                for ($idx = 1; $idx < count($file_contents); $idx++) {
+                    $row = $file_contents[$idx];
+                    echo "<tr>";
+                    foreach ($row as $cell) {
+                        echo "<td>" . $cell . "</td>";
+                    }
+                    echo "</tr>";
+                }
+                ?>
+            </tbody>
+        </table>
+    </div>
+    
+    <!-- ko if: mode()=='preview' -->
+    <div id="preview" data-bind="with: preview_data">
+        <h1><?php echo xlt('Preview Changes'); ?></h1>
+        <input type="button" id="preview" value="<?php echo xla('Commit Changes'); ?>" data-bind="click: commitChanges"></input>        
+        <div>
+            <span><?php echo xlt('Unchanged Entries verified'); ?>:</span><span data-bind="text:unchanged()"></span>
+        </div>
+        <div>
+            <span><?php echo xlt('Empty Definitions'); ?>:</span><span data-bind="text:empty()"></span>
+        </div>
+        <div>
+            <div><?php echo xlt('Created Definitions'); ?>:</span><span data-bind="text:created().length"></div>
+            <div><?php echo xlt('Updated Definitions'); ?>:</span><span data-bind="text:updated().length"></div>
+            <h3><?php echo xlt('Created Definitions List'); ?></h3>
+            <div data-bind="foreach: created">
+                <div data-bind="text:$data"></div>
+            </div>
+            <h3><?php echo xlt('Updated Definitions List'); ?></h3>
+            <div data-bind="foreach: updated">
+                <div data-bind="text:$data"></div>
+            </div>
+        </div>
+    </div>
+    <!-- /ko -->
+
+    <!-- ko if: mode()=='committed' -->
+    <div id="committed" data-bind="with: review_data">
+        <h1><?php echo xlt('Committed Changes'); ?></h1>
+        <div>
+            <span><?php echo xlt('Unchanged Entries verified'); ?>:</span><span data-bind="text:unchanged()"></span>
+        </div>
+        <div>
+            <span><?php echo xlt('Empty Definitions'); ?>:</span><span data-bind="text:empty()"></span>
+        </div>
+        <div>
+            <div><?php echo xlt('Created Definitions'); ?>:</span><span data-bind="text:created().length"></div>
+            <div><?php echo xlt('Updated Definitions'); ?>:</span><span data-bind="text:updated().length"></div>
+            <h3><?php echo xlt('Created Definitions List'); ?></h3>
+            <div data-bind="foreach: created">
+                <div data-bind="text:$data"></div>
+            </div>
+            <h3><?php echo xlt('Updated Definitions List'); ?></h3>
+            <div data-bind="foreach: updated">
+                <div data-bind="text:$data"></div>
+            </div>
+        </div>
+    </div>
+    <!-- /ko -->
+
+    <!-- ko if:loading() -->
+        <span data-bind="text:processingStatus"></span><img src='<?php echo $webroot . "/interface/pic/ajax-loader.gif"; ?>'/>
```

Comment:
would be nice if did this with a fontawesome icon rather than the gif

Reply:
Do you know of something else doing that? Maybe I can clone some code....

Reply:
try to replace the img tag with this:
```
<i class="fa fa-spinner fa-spin"></i>
```